text/0000-or-patterns.md

```diff
@@ -0,0 +1,666 @@
+       match self {
+           | Tile::Suited(Suit::Souzu, 2 | 3 | 4 | 6 | 8)
+           | Tile::Dragon(Dragon::Green) => true,
+           _ => false,
+       }
+   }
+   ```
+
+6. Some further examples found with sourcegraph include:
+
+   From [cc-rs](https://sourcegraph.com/github.com/alexcrichton/cc-rs/-/blob/src/lib.rs#L1307:18):
+
+   ```rust
+    match (self.cpp_set_stdlib.as_ref(), cmd.family) {
+        (None, _) => {}
+        (Some(stdlib), ToolFamily::Gnu | ToolFamily::Clang) => {
+            cmd.push_cc_arg(format!("-stdlib=lib{}", stdlib).into());
+        }
+        _ => {
+            ...
+        }
+    }
+   ```
+
+   From [capnproto](https://sourcegraph.com/github.com/capnproto/capnproto-rust/-/blob/capnp/src/private/layout.rs#L1979-2002):
+
+   ```rust
+   // Check whether the size is compatible.
+   match expected_element_size {
+       None | Some(Void | InlineComposite) => (),
+       Some(Bit) => { ... }
+       Some(Byte | TwoBytes | FourBytes | EightBytes) => { ... },
+       ...
+   }
+   ```
+
+   From [chrono](https://sourcegraph.com/github.com/chronotope/chrono/-/blob/src/format/parsed.rs#L271-308):
+
+   ```rust
+   fn resolve_year(y: Option<i32>, q: Option<i32>,
+                   r: Option<i32>) -> ParseResult<Option<i32>> {
+       match (y, q, r) {
+           (y, None, None) => Ok(y),
+           (Some(y), q, r @ (Some(0...99) None)) => { ... },
+           ...
+       }
+   }
+   ```
+
+   From maidsafe's [routing](https://sourcegraph.com/github.com/maidsafe/routing/-/blob/src/states/node.rs#L2138-2180):
+
+   ```rust
+   match self.peer_mgr.connection_info_received(...) {
+       ...,
+       Ok(IsProxy | IsClient | IsJoiningNode) => { ... },
+       Ok(Waiting | IsConnected) | Err(_) => (),
+   }
+   ```
+
+   Also from [routing](https://sourcegraph.com/github.com/maidsafe/routing/-/blob/src/states/node.rs#L2215-2245):
+
+   ```rust
+        match self.peer_mgr.connection_info_received(...) {
+            Ok(Ready(our_info, their_info)) => { ... }
+            Ok(Prepare(_) | IsProxy | IsClient | IsJoiningNode) => { ... }
+            Ok(Waiting | IsConnected) | Err(_) => (),
+        }
+
+   ```
+
+   From [termion](https://sourcegraph.com/github.com/ticki/termion/-/blob/src/input.rs#L143-153):
+
+   ```rust
+   for c in self.bytes() {
+       match c {
+           Err(e) => return Err(e),
+           Ok(0 | 3 | 4) => return Ok(None),
+           Ok(0x7f) => { buf.pop(); }
+           Ok(b'\n' | b'\r') => break,
+           Ok(c) => buf.push(c),
+       }
+   }
+   ```
+
+7. Some other use cases are:
+
+   From git2-rs:
+
+   ```rust
+   match obj.kind() {
+       Some(Commit | Tag | Tree) => ...
+       Some(Blob) => ...
+       None => ...
+   }
+   ```
+
+   From [debcargo](https://salsa.debian.org/rust-team/debcargo/blob/master/src/debian/dependency.rs#L234):
+
+   ```rust
+   match (op, &mmp.clone()) {
+       (&Lt, &M(0) | &MM(0, 0) | &MMP(0, 0, 0)) => debcargo_bail!(
```

Comment:
Could be simplified slightly more to `(&Lt, &(M(0) | MM(0, 0) | MMP(0, 0, 0)))` (and similar for the other branches).